awscli/customizations/s3/subcommands.py

```diff
@@ -1008,6 +1024,37 @@ def run(self):
             rc = 2
         return rc
 
+    def _get_s3_transfer_handler(self, result_queue):
```

Comment:
Seems like it's worth pulling this into a separate class, both for testability, but also as a clean slate to start integrating more and more of the s3transfer logic over.